eclair-core/src/main/scala/fr/acinq/eclair/payment/Bolt12Invoice.scala

```diff
@@ -39,57 +39,34 @@ case class Bolt12Invoice(records: TlvStream[InvoiceTlv]) extends Invoice {
 
   import Bolt12Invoice._
 
-  val amount: MilliSatoshi = records.get[Amount].map(_.amount).get
+  val invoiceRequest: InvoiceRequest = InvoiceRequest.validate(TlvStream[InvoiceRequestTlv](records.records.collect { case tlv: InvoiceRequestTlv => tlv }, records.unknown.filter(_.tag < UInt64(160)))).toOption.get
+
+  val amount: MilliSatoshi = records.get[InvoiceAmount].map(_.amount).get
   override val amount_opt: Option[MilliSatoshi] = Some(amount)
-  override val nodeId: Crypto.PublicKey = records.get[NodeId].get.publicKey
-  override val paymentHash: ByteVector32 = records.get[PaymentHash].get.hash
-  override val description: Either[String, ByteVector32] = Left(records.get[Description].get.description)
-  override val createdAt: TimestampSecond = records.get[CreatedAt].get.timestamp
-  override val relativeExpiry: FiniteDuration = FiniteDuration(records.get[RelativeExpiry].map(_.seconds).getOrElse(DEFAULT_EXPIRY_SECONDS), TimeUnit.SECONDS)
-  override val minFinalCltvExpiryDelta: CltvExpiryDelta = records.get[Cltv].map(_.minFinalCltvExpiry).getOrElse(DEFAULT_MIN_FINAL_EXPIRY_DELTA)
-  override val features: Features[InvoiceFeature] = records.get[FeaturesTlv].map(_.features.invoiceFeatures()).getOrElse(Features.empty)
-  val chain: ByteVector32 = records.get[Chain].map(_.hash).getOrElse(Block.LivenetGenesisBlock.hash)
-  val offerId: Option[ByteVector32] = records.get[OfferId].map(_.offerId)
-  val blindedPaths: Seq[RouteBlinding.BlindedRoute] = records.get[Paths].get.paths
-  val blindedPathsInfo: Seq[PaymentInfo] = records.get[PaymentPathsInfo].get.paymentInfo
-  val issuer: Option[String] = records.get[Issuer].map(_.issuer)
-  val quantity: Option[Long] = records.get[Quantity].map(_.quantity)
-  val refundFor: Option[ByteVector32] = records.get[RefundFor].map(_.refundedPaymentHash)
-  val payerKey: Option[ByteVector32] = records.get[PayerKey].map(_.publicKey)
-  val payerNote: Option[String] = records.get[PayerNote].map(_.note)
-  val payerInfo: Option[ByteVector] = records.get[PayerInfo].map(_.info)
-  val fallbacks: Option[Seq[FallbackAddress]] = records.get[Fallbacks].map(_.addresses)
-  val refundSignature: Option[ByteVector64] = records.get[RefundSignature].map(_.signature)
-  val replaceInvoice: Option[ByteVector32] = records.get[ReplaceInvoice].map(_.paymentHash)
+  override val nodeId: Crypto.PublicKey = records.get[InvoiceNodeId].get.nodeId
+  override val paymentHash: ByteVector32 = records.get[InvoicePaymentHash].get.hash
+  override val description: Either[String, ByteVector32] = Left(invoiceRequest.offer.description)
+  override val createdAt: TimestampSecond = records.get[InvoiceCreatedAt].get.timestamp
+  override val relativeExpiry: FiniteDuration = FiniteDuration(records.get[InvoiceRelativeExpiry].map(_.seconds).getOrElse(DEFAULT_EXPIRY_SECONDS), TimeUnit.SECONDS)
+  override val minFinalCltvExpiryDelta: CltvExpiryDelta = CltvExpiryDelta(0)
```

Comment:
That indicates that something should be changed, I don't think a value of `0` makes sense here?

Reply:
What I mean is that if you really don't want to provide that value, it shouldn't be in the shared trait, this code shows that there's an issue in the abstraction here.

Reply:
Either we have a meaningful value to provide here, or we should remove it from the shared abstraction because it's actually not something that makes sense for both Bolt11 and Bolt12 invoices.

Reply:
Done